packages/react-instantsearch-hooks/src/components/__tests__/DynamicWidgets.test.tsx

```diff
@@ -423,8 +421,8 @@ describe('DynamicWidgets', () => {
       rerender(<App attributes={['brand', 'categories']} />);
     });
 
-    await wait(0);
     await waitFor(() => {
+      expect(searchClient.search).toHaveBeenCalledTimes(3);
```

Comment:
It makes a bit more sense to "guard' the snapshots with the `waitFor` like this, wdyt?

```ts
    await waitFor(() => {
      expect(searchClient.search).toHaveBeenCalledTimes(1);
    });

    expect(container).toMatchInlineSnapshot(`
<div>
  RefinementList(brand)
</div>
`);

    expect(indexContextRef.current!.getWidgets()).toEqual([
      expect.objectContaining({ $$type: 'ais.refinementList' }),
      expect.objectContaining({ $$type: 'ais.dynamicWidgets' }),
    ]);

    act(() => {
      rerender(<App attributes={['brand', 'categories']} />);
    });

    await waitFor(() => {
      expect(searchClient.search).toHaveBeenCalledTimes(3);
    });

    expect(container).toMatchInlineSnapshot(`
<div>
  RefinementList(brand)
  Menu(categories)
</div>
`);

    expect(indexContextRef.current!.getWidgets()).toEqual([
      expect.objectContaining({ $$type: 'ais.refinementList' }),
      expect.objectContaining({ $$type: 'ais.dynamicWidgets' }),
      expect.objectContaining({ $$type: 'ais.menu' }),
    ]);

    act(() => {
      rerender(<App attributes={['brand']} />);
    });

    await waitFor(() => {
      expect(searchClient.search).toHaveBeenCalledTimes(5);
    });

    expect(container).toMatchInlineSnapshot(`
<div>
  RefinementList(brand)
</div>
`);
```